### ISSUE TITLE
Admin Lockout after upgrade from 1.7.8.9 to 8.0.4

### ISSUE THREAD

**Issue by mrkalchemy** [PrestaShop/PrestaShop#32709]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

After upgrading an existing 1.7.8.9 install to 8.0.4 the admin web interface returns an error and a log entry is added stating that PREFIX_employee_session.date_add is invalid:

[2023-05-25 10:04:07] request.CRITICAL: Uncaught PHP Exception PrestaShopException: "Property EmployeeSession-&gt;date_add is not valid" at /.../classes/ObjectModel.php line 1085 {"exception":"[object] (PrestaShopException(code: 0): Property EmployeeSession-&gt;date_add is not valid at /.../classes/ObjectModel.php:1085)"} []

This appears to be caused by the lack of a default value for this column and something failing catastrophically when it sees a zero date (0000-00-00).

### Expected behavior

Should upgrade sessions gracefully or delete them all.

Option 1:  Fix the bad dates
e.g. 
UPDATE `PREFIX_employee_session` SET `date_add` = NOW() WHERE `date_add` IS NULL OR `date_add` = '0000-00-00';
UPDATE `PREFIX_employee_session` SET `date_upd` = NOW() WHERE `date_upd` IS NULL OR `date_upd` = '0000-00-00';
UPDATE `PREFIX_customer_session` SET `date_add` = NOW() WHERE `date_add` IS NULL OR `date_add` = '0000-00-00';
UPDATE `PREFIX_customer_session` SET `date_upd` = NOW() WHERE `date_upd` IS NULL OR `date_upd` = '0000-00-00';

Option 2: Truncate the PREFIX_employee_session table to allow admins to get to the login page without having to manually delete their cookies.

### Steps to reproduce

Set up a server using Apache, PHP 7.4 and MariaDB 8.0
Check to make sure MariaDB is set-up to allow zero dates in DATETIME fields.
Restore a backup of an existing install of PrestaShop 1.7.8.9 or do a clean install.
Upgrade the existing 1.7.8.9 install to 8.0.4 using the autoupgrade module using a local zip and xml file.
Once the upgrade is complete click on the dashboard link and you will see the error.

### PrestaShop version(s) where the bug happened

from 1.7.8.9 to 8.0.4

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

autoupgrade  v4.15.0

**Comment by mrkalchemy** [comment]
We have submitted a pull request that mitigates this issue by replacing zero dates with the current date.
https://github.com/PrestaShop/autoupgrade/pull/584

**Comment by hibatallahAouadni** [comment]
Hello @mrkalchemy 

Thanks for reporting the issue!
Ping @PrestaShop/prestashop-core-developers someone please try to reproduce the issue, it's too technical!

Thanks in advance!

**Comment by boherm** [comment]
Unfortunately, we have not been able to reproduce the issue described in several configurations. 😅
Could be possible to provide us more informations? 🤔

(Without news from you, we would be forced to close this issue within 2 weeks.)

**Comment by mrkalchemy** [comment]
Okay here's a summary including as much detail as we can.

We discovered the problem on our local test server while checking the upgrade path to the **8.0.x** series from a restored backup of our existing **1.7.8.9** installation. The test server was running **Ubuntu server 22.04.2 LTS** with **PHP 7.4.33** and the default **MySQL version "8.0.33-0ubuntu0.22.04.2 - (Ubuntu)"**. The MySQL **'sql_mode'** was set to **"STRICT_TRANS_TABLES,​NO_ENGINE_SUBSTITUTION"** and did not include **'NO_ZERO_DATE'** or **'NO_ZERO_IN_DATE'**.

During the upgrade the `_employee_session` table was modified to add two date fields that did not include a default value and this resulted in them being populated with '0000-00-00 00:00:00' values for existing records. The database server did not raise an error as it was configured to allow this. Unfortunately after the upgrade was completed the back office showed an error and it was imposable to login without manually deleting the session cookies from the web browser. This lockout appears to have been caused by the `_employee_session` record, associated with the session cookie, containing an invalid date and triggering an Exception from the ObjectModel subclass.

We made a patch to fix the issue (see link above) that replaces the zero dates with a valid value during the upgrade process.

If you have any specific questions do let us know.

Thanks!

**Comment by mrkalchemy** [comment]
We've done some more testing and the problem may be more complex than we originally thought. The PR we submitted does solve the problem but it seems that the issue does not occur for every installation.

We tried upgrading a fresh install of 1.7.8.9 but that seemed to have different issues that we believe are being addressed elsewhere.

So then we tried going from a clean install of 1.7.7.8 to 1.7.8.9 and then finally to 8.0.4. This did not trigger the bug and admin login was possible.

The original install that triggered the bug is quite old and has been through multiple upgrades since it was first setup using version 1.5.4.1 so this may take a while to track down the cause.

We will post an update after we have done some further testing to isolate the issue.

Thanks!
